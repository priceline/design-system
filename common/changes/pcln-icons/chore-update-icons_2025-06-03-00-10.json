{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}