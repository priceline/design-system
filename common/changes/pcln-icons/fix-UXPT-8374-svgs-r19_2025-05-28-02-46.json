{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "replace defaultProps usage in svgr template file",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}