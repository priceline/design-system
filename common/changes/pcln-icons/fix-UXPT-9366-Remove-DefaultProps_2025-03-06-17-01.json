{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "remove defaultProps",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}