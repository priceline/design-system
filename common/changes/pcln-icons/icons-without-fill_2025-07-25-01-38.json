{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "UXPT-10383: rounded Material SVGs from filled to unfilled",
      "type": "major"
    }
  ],
  "packageName": "pcln-icons"
}