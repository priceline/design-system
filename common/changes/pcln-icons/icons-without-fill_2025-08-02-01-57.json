{
  "changes": [
    {
      "comment": "allow styled-components 6 as a peer",
      "type": "patch",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}