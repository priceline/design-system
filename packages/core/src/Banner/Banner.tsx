import {
  Attention as AttentionIcon,
  Information as InformationIcon,
  Success as SuccessIcon,
  Warning as WarningIcon,
} from 'pcln-icons'
import React from 'react'
import styled from 'styled-components'
import { Box, type BoxProps } from '../Box/Box'
import { CloseButton } from '../CloseButton/CloseButton'
import { Flex } from '../Flex/Flex'
import { Text } from '../Text/Text'
import { applyVariations } from '../utils/utils'

/**
 * @public
 */
export type BannerColor = {
  backgroundColor?: string
  color?: string
  icon?: React.ReactElement
}

const bannerColors: Record<string, BannerColor> = {
  green: {
    backgroundColor: 'secondary.base',
    color: 'text.lightest',
    icon: <SuccessIcon />,
  },
  lightGreen: {
    backgroundColor: 'secondary.light',
    color: 'secondary.dark',
    icon: <SuccessIcon />,
  },
  red: {
    backgroundColor: 'error.base',
    color: 'text.lightest',
    icon: <WarningIcon />,
  },
  lightRed: {
    backgroundColor: 'error.light',
    color: 'error.dark',
    icon: <WarningIcon />,
  },
  orange: {
    backgroundColor: 'alert.base',
    color: 'text.lightest',
    icon: <AttentionIcon />,
  },
  blue: {
    backgroundColor: 'primary.base',
    color: 'text.lightest',
    icon: <InformationIcon />,
  },
  lightBlue: {
    backgroundColor: 'primary.light',
    color: 'primary.dark',
    icon: <InformationIcon />,
  },
}

const StyledBox = styled(Box)`
  ${applyVariations('Banner')}
`

/**
 * @public
 */
export type BannerProps = BoxProps & {
  bg?: string
  color?: string
  children?: React.ReactNode
  header?: string | React.ReactNode
  icon?: React.ReactElement
  onClose?: () => void
  showIcon?: boolean
  text?: string | React.ReactNode
  textAlign?: string
}

/**
 * @public
 */
export function Banner({
  bg,
  color = 'green',
  children,
  header,
  icon,
  onClose,
  showIcon = true,
  text,
  textAlign = 'left',
  ...props
}: BannerProps): React.ReactElement {
  const bannerColor = bannerColors[!bg && color === 'green' ? color : bg] || {}
  const Icon = icon || bannerColor.icon
  const finalColor = !bannerColor.color ? color : bannerColor.color

  let finalHeader = null

  if (header) {
    finalHeader = React.isValidElement(header) ? header : <Text textStyle='display2'>{header}</Text>
  }

  return (
    <StyledBox {...props} textAlign={textAlign} bg={bannerColor.backgroundColor || bg} color={finalColor}>
      <Flex justifyContent='space-between' alignItems='flex-start'>
        {!!Icon && !!showIcon && React.cloneElement(Icon, { mr: 2, size: 24, mt: '-2px' })}
        <Box textAlign={textAlign} width={1}>
          {finalHeader}
          <Text.span fontSize={1}>{text}</Text.span>
          {children}
        </Box>
        {!!onClose && <CloseButton onClick={onClose} ml={2} size={24} title='close' mt='-2px' />}
      </Flex>
    </StyledBox>
  )
}
