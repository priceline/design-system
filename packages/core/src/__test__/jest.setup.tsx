import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
