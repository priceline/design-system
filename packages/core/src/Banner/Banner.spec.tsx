import { Star } from 'pcln-icons'
import React from 'react'
import { Banner } from '../Banner/Banner'
import { Heading } from '../Heading/Heading'
import { Text } from '../Text/Text'
import { render } from '../__test__/testing-library'
import { theme } from '../theme/theme'

describe('Banner', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders with no props other than theme', () => {
    const json = rendererCreateWithTheme(<Banner />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with an icon', () => {
    const json = rendererCreateWithTheme(<Banner icon={<Star />} />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with text string', () => {
    const { asFragment } = render(<Banner header='Header' text='Text' icon={<Star />} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with text node', () => {
    const json = rendererCreateWithTheme(
      <Banner header='Header' text={<Text>Text</Text>} icon={<Star />} />
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with blue bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='blue' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with green bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='green' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with orange bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='orange' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with red bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='red' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with lightBlue bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='lightBlue' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightBlue)
    expect(json).toHaveStyleRule('color', theme.colors.darkBlue)
  })

  test('renders with lightGreen bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='lightGreen' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGreen)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('renders with lightRed bg', () => {
    const json = rendererCreateWithTheme(<Banner bg='lightRed' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightRed)
    expect(json).toHaveStyleRule('color', theme.colors.darkRed)
  })

  test('renders close button if onClose func is provided', () => {
    function noop() {}
    const { container } = render(<Banner onClose={noop} />)
    const closeButton = container.querySelector('button')
    expect(closeButton).toBeTruthy()
  })

  test('does render blue left-hand icon by default', () => {
    const { container } = render(<Banner bg='blue' />)
    const icon = container.querySelector('svg')
    // todo: check svg path/attrs for correct icon
    expect(icon).toBeTruthy()
  })

  test('does not render blue left-hand icon if showIcon is false', () => {
    const { container } = render(<Banner bg='blue' showIcon={false} />)
    const icon = container.querySelector('svg')
    expect(icon).toBeFalsy()
  })

  test('accepts non-preset colors', () => {
    const json = rendererCreateWithTheme(<Banner bg='gray' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.gray)
  })

  test('renders content from children props', () => {
    const json = rendererCreateWithTheme(
      <Banner>
        <span>123</span>
      </Banner>
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with header node', () => {
    const { asFragment, getByText } = render(
      <Banner header={<Heading>Hello world</Heading>} text={<Text>Text</Text>} />
    )

    getByText(/Hello world/)
    expect(asFragment()).toMatchSnapshot()
  })
})
