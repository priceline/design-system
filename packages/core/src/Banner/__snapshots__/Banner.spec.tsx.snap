// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner accepts non-preset colors 1`] = `
.c0 {
  background-color: #4f6f8f;
  color: #0a0;
  text-align: left;
}

.c2 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  font-size: 14px;
}

<div
  className="c0 "
  color="green"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      width={1}
    >
      <span
        className="c3"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders content from children props 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
      <span>
        123
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner renders with an icon 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17.8l6.2 3.7-1.6-7L22 9.7l-7.2-.6L12 2.5 9.2 9.1 2 9.7l5.5 4.8-1.6 7 6.1-3.7z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with blue bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #0068ef;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with green bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with header node 1`] = `
<DocumentFragment>
  .c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c6 {
  font-size: 14px;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class="c1 "
      color="text.lightest"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="c3 c4"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
          />
        </svg>
        <div
          class="c5"
          width="1"
        >
          <h3
            class=""
          >
            Hello world
          </h3>
          <span
            class="c6"
            font-size="1"
          >
            <div
              class=""
            >
              Text
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner renders with lightBlue bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #e8f2ff;
  color: #049;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="primary.dark"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightGreen bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #ecf7ec;
  color: #060;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="secondary.dark"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightRed bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #fbebeb;
  color: #800;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="error.dark"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 11h-2V7h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with no props other than theme 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with orange bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #f06f20;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 21.5h22l-11-19-11 19zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with red bg 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #c00;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 11h-2V7h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <span
        className="c5"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with text node 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c3 {
  outline: none;
}

.c0 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c4 {
  text-align: left;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
}

.c6 {
  font-size: 14px;
}

<div
  className="c0 "
  color="text.lightest"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 17.8l6.2 3.7-1.6-7L22 9.7l-7.2-.6L12 2.5 9.2 9.1 2 9.7l5.5 4.8-1.6 7 6.1-3.7z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <div
        className="c5"
      >
        Header
      </div>
      <span
        className="c6"
        fontSize={1}
      >
        <div
          className=""
        >
          Text
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner renders with text string 1`] = `
<DocumentFragment>
  .c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0a0;
  color: #fff;
  text-align: left;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c6 {
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
}

.c7 {
  font-size: 14px;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class="c1 "
      color="text.lightest"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="c3 c4"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 17.8l6.2 3.7-1.6-7L22 9.7l-7.2-.6L12 2.5 9.2 9.1 2 9.7l5.5 4.8-1.6 7 6.1-3.7z"
          />
        </svg>
        <div
          class="c5"
          width="1"
        >
          <div
            class="c6"
          >
            Header
          </div>
          <span
            class="c7"
            font-size="1"
          >
            Text
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
